csrc/mmdeploy/codebase/mmseg/segment.cpp

```diff
@@ -44,6 +51,29 @@ class ResizeMask : public MMSegmentation {
     OUTCOME_TRY(auto host_tensor, MakeAvailableOnDevice(mask, host, stream_));
     OUTCOME_TRY(stream_.Wait());
 
+    if (!with_argmax_ && mask.shape(1) > 1 && mask.shape(1) == classes_ &&
+        host_tensor.data_type() == DataType::kFLOAT) {
+      int stride = height * width;
+      Tensor mask_out = TensorDesc{Device("cpu"),
```

Comment:
why not output `masdk_out` directly?

Reply:
`struct mmdeploy_segmentation_t` in `segmentor.h` can be defined as,
```c
typedef struct mmdeploy_segmentation_t {
  int height;   ///< height of \p mask that equals to the input image's height
  int width;    ///< width of \p mask that equals to the input image's width
  int classes;  ///< the number of labels in \p mask
  mmdeploy_data_type_t data_type; 
  void* mask;    ///< segmentation mask of the input image, in which mask[i * width + j] indicates
                ///< the label id of pixel at (i, j) if `data_type` is INT32. Otherwisse, mask means the probability feature map with shape as (classes, height, width)
} mmdeploy_segmentation_t;
```

Reply:
> why not output `masdk_out` directly? Let users decide how to do with the probability feature map

That would change other language interfaces like JAVA. I'm not familiar with that though.

Reply:
I see.

Reply:
I think the API has to be changed. A [feishu task](https://project.feishu.cn/openmmlab/story/detail/3325854) has been scheduled.
In this PR, I think logging an error message when `if (!with_argmax_)` is more appropriate.
```c++
if (!with_argmax_) {
    MMDEPLOY_ERROR("TODO: SDK will support probability featmap soon.");
    return Status(eNotSupported);
}
```